File: synchronizinglists.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using std::cin;
using std::cout;
using std::string;

int n, temp;

std::vector<int> list1(n);
std::vector<int> list1_copy(n);
std::vector<int> list2(n);

int main(int argc, char** argv) {
    do {
        cin >> n;
        
        for(int i = 0; i < 2 * n; i++) {
            if(i < n) {
                cin >> temp;
                list1.push_back(temp);
                list1_copy.push_back(temp);
            } else {
                cin >> temp;
                list2.push_back(temp);
            }
        }
        std::sort(list1_copy.begin() , list1_copy.end());
        std::sort(list2.begin() , list2.end());
        
        for(int i = 0; i < list1.size(); i++) {
            for(int j = 0; j < list2.size(); j++) {
                if(list1.at(i) == list1_copy.at(j)) {
                    cout << list2[j] << std::endl;
                    break;
                }
            }
        }
       cout << std::endl;
       
       list1.clear();
       list1_copy.clear();
       list2.clear();
    } while(n != 0);

    return 0;
}

File: synchronizinglists.java
mport java.util.Arrays;
import java.util.Scanner;

public class SyncLists {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n, temp;
        do {
            n = sc.nextInt();
            int[] list1 = new int[n];
            int[] copyList1 = new int[n];
            int[] list2 = new int[n];

            for (int i = 0; i < 2 * n; i++) {
                temp = sc.nextInt();
                
                if (i < n) {
                    list1[i] = temp;
                    copyList1[i] = temp;
                } else {
                    list2[i - n] = temp;
                }
            }

            Arrays.sort(copyList1);
            Arrays.sort(list2);
            

            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    if (list1[i] == copyList1[j])
                        System.out.println(list2[j]);
                }
            }
            
            System.out.println();
        } while(n != 0);
        
    }
}

 rss fee
